GitVersionExe/GitPreparer.cs

```diff
@@ -20,31 +21,74 @@ public bool IsDynamicGitRepository
 
         public string DynamicGitRepositoryPath { get; private set; }
 
-        public string Prepare()
+        public void InitialiseDynamicRepositoryIfNeeded()
         {
-            var gitPath = arguments.TargetPath;
+            if (string.IsNullOrWhiteSpace(arguments.TargetUrl)) return;
 
-            if (!string.IsNullOrWhiteSpace(arguments.TargetUrl))
+            var targetPath = CalculateTemporaryRepositoryPath(arguments.TargetUrl);
+            DynamicGitRepositoryPath = CreateDynamicRepository(targetPath, arguments.Authentication, arguments.TargetUrl, arguments.TargetBranch);
+        }
+
+        string CalculateTemporaryRepositoryPath(string targetUrl)
```

Comment:
I would still love to create the dynamic repo in a custom location, i.e. set via a cmd-line argument.